rust/lancedb/src/query.rs

```diff
@@ -351,6 +352,17 @@ pub trait QueryBase {
     /// on the filter column(s).
     fn only_if(self, filter: impl AsRef<str>) -> Self;
 
+    /// Perform a full text search on the table.
+    ///
+    /// The results will be returned in order of BM25 scores.
+    ///
+    /// This method is only valid on tables that have a full text search index.
+    ///
+    /// ```ignore
+    /// query.full_text_search(FullTextSearchQuery::new("hello world"))
+    /// ```
+    fn full_text_search(self, query: FullTextSearchQuery) -> Self;
```

Comment:
Do we expect that we will eventually allow FTS to be combined with vector search in a single query?  Or will reranking always involve two queries?

Reply:
Now reranking is done in lancedb python, i think we can do FTS and vector search simultaneously in a single query, which is more efficient, but it requires moving these reranker into lance in rust.